src/modules/PublicLab.TitleModule.Related.js

```diff
@@ -25,59 +25,142 @@ Results should be in following JSON format:
+  function showResults(resultContainer, result, resultClass) {
+    resultContainer.find(resultClass + ' .title').html(result.title);
+    resultContainer.find(resultClass + ' .title').attr('href', result.url);
+    resultContainer.find(resultClass + ' .author').html('@' + result.author);
+    resultContainer.find(resultClass + ' .author').attr('href', '/profile/' + result.author);
+  }
+
+  function showAdded(addedResult, relatedResults) {
+    addedRelatedEl.append('<div class="addedresult addedresult-' + addedResult.id + '" style="margin: 3px;"><a class="btn btn-outline-secondary btn-sm remove-tag"><i class="fa fa-times-circle"></i> Remove</a> <a class="title"></a> by <a class="author"></a></div>');
+    var addedResultClass = '.addedresult-' + addedResult.id;
+    showResults(addedRelatedEl, addedResult, addedResultClass);
+
+    // For removing tags
+    $('.addedresult-' + addedResult.id + ' .remove-tag').click(function() {
+      var selectedToken = (editor.tagsModule.el.find('.token[data-value="response:' + addedResult.id + '"]'));
+      var y_axis = window.scrollY;
+      selectedToken.find('.close').trigger('click');
+      window.scrollTo(window.scrollX, y_axis);
+      addedRelatedPost = addedRelatedPost.replace(addedResult.id, ''); // remove tags from tagIntergrationModule.
+      $('.addedresult-' + addedResult.id).remove();
+      showRelatedResult(relatedResults);
+      if (addedRelatedPost === 'addedTitles') {
+        $('.related-post').remove(); // removing the tag from Realted Post section.
+      }
+    });
   }
 
   // expects array of results in format:
   // { id: 3, title: 'A third related post', url: '/', author: 'bsugar'}
-  function show(relatedResults) {
+  function showRelatedResult(relatedResults) {
     relatedEl.find('.result').remove();
+    // If tag removed from TagModule
+    editor.tagsModule.el.find('input').on('tokenfield:removedtoken', function(e) {
+      var id = (e.attrs.value).replace('response:', '');
+      if (addedRelatedPost.includes(id)) {
+        addedRelatedPost = addedRelatedPost.replace(id, '');
+        $('.addedresult-' + id).remove();
+        showRelatedResult(relatedResults);
+        if (addedRelatedPost === 'addedTitles') {
+          $('.related-post').remove();
+        }
+      }
+    });
 
     relatedResults.slice(0, 8).forEach(function(result) {
-      relatedEl.append('<div class="result result-' + result.id + '" style="margin: 3px;"><a class="btn btn-xs btn-default add-tag"><i class="fa fa-plus-circle"></i> Add</a> <a class="title"></a> by <a class="author"></a></div>');
-      relatedEl.find('.result-' + result.id + ' .title').html(result.title);
-      relatedEl.find('.result-' + result.id + ' .title').attr('href', result.url);
-      relatedEl.find('.result-' + result.id + ' .author').html('@' + result.author);
-      relatedEl.find('.result-' + result.id + ' .author').attr('href', '/profile/' + result.author);
+      var showRelated = false;
```

Comment:
No need for this variable, directly use if condition of line 93 without negation